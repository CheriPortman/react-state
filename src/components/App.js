import React, { PureComponent } from 'react';
import ColorPicker from './ColorPicker';
import ColorDisplay from './ColorDisplay';
import Box from './Box';
import BoxLooksPicker from './BoxLooksPicker';


export default class App extends PureComponent {
  state = {
    color: '',
    boxColor: 'white'
  }

  selectColor = color => {
    this.setState({ color });
  }

  selectBox = boxColor => {
    this.setState({ boxColor });
  }

  render() {
    const { color } = this.state;
    const { boxColor } = this.state;
    // eslint-disable-next-line
    console.log({ color });

    return (
      <>
        <ColorPicker selectColorHandler={this.selectColor} />
        <ColorDisplay color={color} />
        <h2>Click to see the color:</h2>
        <BoxLooksPicker selectBoxHandler={this.selectBox} />
        <Box boxColor={boxColor} />
      </>
    );
    
  }
}
