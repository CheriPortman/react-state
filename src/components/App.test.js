import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import { exportAllDeclaration } from '@babel/types';

describe('App componenet', () => {
  it('renders App', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
