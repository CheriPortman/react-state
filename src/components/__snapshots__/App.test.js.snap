// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App componenet renders App 1`] = `<RandomColor />`;
